src/gcodePlanner.cpp

```diff
@@ -299,7 +299,12 @@ GCodePath& GCodePlanner::addTravel(Point p)
                 }
                 path = getLatestPathWithConfig(&travel_config, SpaceFillType::None);
                 path->retract = retract;
-                // don't perform a z-hop
+                if (always_retract)
+                {
+                    // do a z-hop (if generally enabled)
+                    // not sure yet if this is a good or a bad thing to do
+                    path->perform_z_hop = perform_z_hops;
```

Comment:
I don't think we should do this. The z hop is not done because of the retraction, but was generally applied in the same situations. I don't think the forcing of retraction should have influence on when we do a z hop.

Moreover, if we would have done this it should have been `perform_z_hops && !perform_z_hops_only_when_collides`.

Please remove this logic.